pages/admin-dashboard.js

```diff
@@ -1,32 +1,39 @@
-import { useState, useEffect } from 'react';
+import { useState } from 'react';
 import { allProfilesInfo } from '../graphql/api';
 import Profile from '../components/admin-dashboard/profile';
-
-const getProfiles = (data) => {
-  return data ? data.allProfilesInfo.data : [];
-};
+import useUser from '../hooks/useUser';
 
 export default function AdminDashboard() {
-  const { data, errorMessage } = allProfilesInfo();
+  const user = useUser();
   const [profiles, setProfiles] = useState([]);
+  const [errorMessage, setErrorMessage] = useState(null);
 
-  useEffect(() => {
-    if (!profiles.length) {
-      setProfiles(getProfiles(data))
+  const getProfiles = async () => {
+    const { data, errorMessage } = await allProfilesInfo();
+    if (!!errorMessage) {
+      setErrorMessage(errorMessage);
+    }
+    else {
+      setProfiles(data ? data.allProfilesInfo.data : []);
     }
-  }, [data, profiles.length]);
+  };
 
-  return (
+  return (user?.role === "Administrator") ? (
```

Comment:
we may need a loading state to show while user is being fetched, otherwise there's a flash of the 403 message.